### PR TITLE
MULE-12243: JMS should publish messages with String Representable Mim…

### PR DESCRIPTION
**Pull request by estebanwasinger**
…eTypes as StringMessages

### CODE REVIEW

File: src/main/java/org/mule/extensions/jms/internal/message/JmsMessageUtils.java [hunk 1]
```diff
@@ -404,4 +395,15 @@ private static boolean validateMapMessageType(Map<?, ?> candidate) {
...
   }
 
+  /**
+   * Utility method which given a {@link MediaType} will verify if this one is {@link String} representable
+   *
+   * @param mediaType to verify
+   * @return boolean indicating whether the {@link MediaType} is String representable or not
+   */
+  private static boolean isStringRepresentable(MediaType mediaType) {
+    return mediaType.getPrimaryType().equals(TEXT) || STRING_REPRESENTABLE_MIME_TYPES
```
Comment (marianogonzalez) [inline src/main/java/org/mule/extensions/jms/internal/message/JmsMessageUtils.java]: this logic should be in MediaTypeUtils

File: src/main/java/org/mule/extensions/jms/internal/message/JmsMessageUtils.java [hunk 2]
```diff
@@ -50,8 +53,11 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(JmsMessageUtils.class);
   private static final char REPLACEMENT_CHAR = '_';
+  private static final String TEXT = "text";
```
Comment (marianogonzalez) [inline src/main/java/org/mule/extensions/jms/internal/message/JmsMessageUtils.java]: you have the exact same constant in MediaTypeUtils

File: src/main/java/org/mule/extensions/jms/internal/message/JmsMessageUtils.java [hunk 3]
```diff
@@ -65,14 +71,16 @@ public static Message toMessage(Object object, Session session) throws JMSExcept
       return mapToMessage((Map<?, ?>) object, session);
 
     } else if (object instanceof InputStream) {
-      return inputStreamToMessage((InputStream) object, session);
-
+      if (isStringRepresentable(typedValueObject.getDataType().getMediaType())) {
+        return stringToMessage(IOUtils.toString((InputStream) object), session);
```
Comment (marianogonzalez) [inline src/main/java/org/mule/extensions/jms/internal/message/JmsMessageUtils.java]: static import

File: src/main/java/org/mule/extensions/jms/internal/message/JmsMessageUtils.java [hunk 4]
```diff
@@ -65,14 +71,16 @@ public static Message toMessage(Object object, Session session) throws JMSExcept
...
+      } else {
+        return byteArrayToMessage(IOUtils.toByteArray((InputStream) object), session);
```
Comment (marianogonzalez) [inline src/main/java/org/mule/extensions/jms/internal/message/JmsMessageUtils.java]: static import

File: src/main/java/org/mule/extensions/jms/internal/message/JmsMessageUtils.java (hunk 3 above)
Comment (estebanwasinger) [inline src/main/java/org/mule/extensions/jms/internal/message/JmsMessageUtils.java]: `toString` is already taken by `Object.toString()`

File: src/main/java/org/mule/extensions/jms/internal/message/JmsMessageUtils.java (hunk 4 above)
Comment (estebanwasinger) [inline src/main/java/org/mule/extensions/jms/internal/message/JmsMessageUtils.java]: `toByteArray ` is already taken by `JmsMessageUtils.toByteArray()`

**Pull request merged**